index.js

```diff
@@ -8,29 +8,41 @@ const defaults = {
 	v4: '127.0.0.1'
 };
 
-function internalIp(family) {
-	return defaultGateway[family]().then(result => {
-		const interfaces = os.networkInterfaces();
-		const gatewayIp = ipaddr.parse(result.gateway);
-		let ret;
-
-		// Look for the matching interface in all local interfaces
-		Object.keys(interfaces).some(name => {
-			return interfaces[name].some(addr => {
-				const prefix = ipaddr.parse(addr.netmask).prefixLengthFromSubnetMask();
-				const net = ipaddr.parseCIDR(`${addr.address}/${prefix}`);
-
-				if (net[0].kind() === gatewayIp.kind() && gatewayIp.match(net)) {
-					ret = net[0].toString();
-				}
-
-				return Boolean(ret);
-			});
+function match(gateway, family) {
```

Comment:
maybe `findIp` might be a better name.

Reply:
from line 16 `// Look for the matching interface in all local interfaces` - that's why I chose the name. doesn't much matter to me what you think it should be, happy to change it, but I thought it was appropriate given that comment

Reply:
Yeah, I still think `findIp` is more appropriate. The matching is only part of the process.

Reply:
aye, will change